pkg/mcp/types.go

```diff
@@ -21,12 +20,25 @@ type Config struct {
 }
 
 type ServerConfig struct {
-	gmcp.ServerConfig `json:",inline"`
-	Files             []File        `json:"files"`
-	ContainerImage    string        `json:"containerImage"`
-	ContainerPort     int           `json:"containerPort"`
-	ContainerPath     string        `json:"containerPath"`
-	Runtime           types.Runtime `json:"runtime"`
+	DisableInstruction bool     `json:"disableInstruction"`
+	Command            string   `json:"command"`
+	Args               []string `json:"args"`
+	Env                []string `json:"env"`
+	URL                string   `json:"url"`
+	Headers            []string `json:"headers"`
+	// Scope ensures separate deployments for each user and project.
+	Scope string `json:"scope"`
+	// ClientScope is used to scope clients to sessions.
+	// This value will not be used to determine if a new deployment of the server is needed, but is used to ensure the
+	// client is new for the session.
+	ClientScope  string   `json:"clientScope"`
+	AllowedTools []string `json:"allowedTools"`
```

Comment:
These fields, except for `ClientScope` are pulled from GPTScript. I also added `ClientScope` here.